scripts/checkMesh/checks/collocated_nodes.py

```diff
@@ -0,0 +1,53 @@
+from collections import defaultdict
+from dataclasses import dataclass
+import logging
+from typing import Tuple
+
+import numpy
+
+import vtk  # TODO use new pyvtk style + deal with ImportError
+
+
+@dataclass(frozen=True)
+class Options:
+    tolerance: float
+
+
+@dataclass(frozen=True)
+class Result:
+    nodes_buckets: Tuple[Tuple[int]]
+
+
+def check(vtk_input_file: str, options: Options) -> Result:
+    reader = vtk.vtkXMLUnstructuredGridReader()  # TODO Find a generic way to read the vtk mesh.
+    reader.SetFileName(vtk_input_file)
+    reader.Update()
+    mesh = reader.GetOutput()
+
+    points = mesh.GetPoints()
+
+    locator = vtk.vtkIncrementalOctreePointLocator()
+    locator.SetTolerance(options.tolerance)
+    output = vtk.vtkPoints()
+    locator.InitPointInsertion(output, points.GetBounds())
+
+    # original ids to/from filtered ids.
+    # original_to_filtered = numpy.ones(points.GetNumberOfPoints(), dtype=int) * -1
+    filtered_to_original = numpy.ones(points.GetNumberOfPoints(), dtype=int) * -1
+
+    rejected_points = defaultdict(list)
+    point_id = vtk.reference(0)
+    for i in range(points.GetNumberOfPoints()):
+        is_inserted = locator.InsertUniquePoint(points.GetPoint(i), point_id)
```

Comment:
I'm curious to see how well this performs on large meshes. Intuitively I'd think crossing Python/C++ boundaries multiple times inside a hot loop is not a viable approach, but happy to be proven wrong.

Reply:
Hell @klevzoff thanks for your comment.

I've tested the function on a ~6.5 million nodes mesh and it lasts ~7.3 seconds on a 2013 CPU.
Difficult to tell what's the influence of the python/c++ bridge.

- The 6.5M _empty_ loop uses 0.1 sec in python (doing nothing, just `pass`).
- The same loop calling a simple `virtual` function from python (like `locator.GetNumberOfNodes()` which returns an `int`) takes 0.55 secondes.

So I assume that if we have two python/c++ calls in this loop, maybe 1 of the 7.3 seconds is used in python/c++ transfer? Not a very smart estimate though.